org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java

```diff
@@ -110,7 +111,11 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
 					final var params = new PrepareRenameParams();
 					params.setTextDocument(identifier);
 					params.setPosition(LSPEclipseUtils.toPosition(offset, document));
-					prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
+					try {
+						prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
+					} catch (TimeoutException e) {
+						LanguageServerPlugin.logInfo("Could not prepare rename due to timeout after 1 seconds in `textDocument/prepareRename`. 'newName' will be used"); //$NON-NLS-1$

```

Comment:
This should be a warning as it's abnormal behavior.

Reply:
Okay, I have changed it. It is only that this warnings with an exception attached look a bit scary on the console, but that is another topic.